akka-stream/src/main/scala/akka/stream/scaladsl/Restart.scala

```diff
@@ -348,14 +356,17 @@ private abstract class RestartWithBackoffLogic[S <: Shape](
     sourceOut
   }
 
-  // Set a timer to restart after the calculated delay
-  protected final def scheduleRestartTimer() = {
+  protected final def maxRestartsReached() = {
```

Comment:
Style protip: No need for the empty parameter list here.

Reply:
Thanks for the feedback.

I decided explicitly to have them here to give a hint that this function has a side effect (resetting the restartCount). I would find it even more misleading if the parameter list is omitted.

@hseeberger Are you fine with that?

Reply:
It doesn't have a side-effect, it's a mere getter (returning a proper value), right? 
Getter-like methods should be without parantheses.

Reply:
Since it mutates a variable outside of its scope (`restartCount`), I was considering this function as "side-effectful".

Anyways I have no strong opinions on this and I trust your judgments, I just removed the parameter list.